pkg/atconf/file_writer.go

```diff
@@ -293,13 +293,28 @@ func (f *FileWriter) getNextNodeNumber() int {
 				if e2 != nil {
 					continue
 				}
-				if i >= nextNodeNumber {
-					nextNodeNumber = i + 1
+				if i >= len(nodes) {
+					nodes = append(nodes, make([]bool, i-len(nodes)+1)...)
 				}
+				nodes[i] = true
 			}
 		}
 	}
-	return nextNodeNumber
+	return nodes
+}
+
+// getNextNodeNumber returns the number to use for the next vertica node.  It
+// determines this by finding a free spot in the nodes slice.
+func (f *FileWriter) getNextNodeNumber(nodes []bool) (nextNodeNumber int, newNodes []bool) {
+	for i := 0; i < len(nodes); i++ {
+		if !nodes[i] {
```

Comment:
I don't understand what's going on here. Same for `buildNodeSlice()`. I get what you are trying to do, but not the implementation.

Reply:
We are picking the next available node number based on nodes[]. We update this slice as we go, which may involve extending it, so we always return a new copy of nodes[].

Any suggestions on how I can make it easier to understand?

Reply:
I think some "why" comments in `buildNodeSlice()` and `getNextNodeNumber()` would help. Especially where you extend the slice. And maybe adding, in `buildNodeSlice() description`, what we are going to do with the slice.